sentry/src/main/java/io/sentry/instrumentation/file/SentryFileInputStream.java

```diff
@@ -0,0 +1,197 @@
+    delegate = data.delegate;
+  }
+
+  private SentryFileInputStream(
+    @NotNull FileInputStreamInitData data
+  ) throws FileNotFoundException {
+    super(data.file);
+    currentSpan = data.span;
+    file = data.file;
+    delegate = data.delegate;
+  }
+
+  private static FileInputStreamInitData init(
+    @Nullable File file,
+    @Nullable FileInputStream delegate
+  ) throws FileNotFoundException {
+    final ISpan span = startSpan();
+    if (delegate == null) {
+      delegate = new FileInputStream(file);
+    }
+    return new FileInputStreamInitData(file, span, delegate);
+  }
+
+  private static FileInputStreamInitData init(
+    @NotNull FileDescriptor fd,
+    @Nullable FileInputStream delegate
+  ) {
+    final ISpan span = startSpan();
+    if (delegate == null) {
+      delegate = new FileInputStream(fd);
+    }
+    // TODO: it's only possible to get filename from FileDescriptor via reflection AND when it's
+    // TODO: running on Android, should we do that?
+    return new FileInputStreamInitData(null, span, delegate);
+  }
+
+  private static @Nullable
+  ISpan startSpan() {
+    final ISpan parent = Sentry.getSpan();
+    return parent != null ? parent.startChild("file.read") : null;
+  }
+
+  @Override
+  public int read() throws IOException {
+    try {
+      int result = delegate.read();
+      if (result != -1) {
+        byteCount++;
+      }
+      return result;
+    } catch (IOException exception) {
+      spanStatus = SpanStatus.INTERNAL_ERROR;
+      throw exception;
+    }
+  }
+
+  @Override
+  public int read(byte @NotNull [] b) throws IOException {
+    try {
+      int result = delegate.read(b);
+      if (result != -1) {
+        byteCount += result;
+      }
+      return result;
+    } catch (IOException exception) {
+      spanStatus = SpanStatus.INTERNAL_ERROR;
+      throw exception;
+    }
+  }
+
+  @Override
+  public int read(byte @NotNull [] b, int off, int len) throws IOException {
+    try {
+      int result = delegate.read(b, off, len);
+      if (result != -1) {
+        byteCount += result;
+      }
+      return result;
+    } catch (IOException exception) {
+      spanStatus = SpanStatus.INTERNAL_ERROR;
+      throw exception;
+    }
+  }
+
+  @Override
+  public long skip(long n) throws IOException {
+    try {
+      long result = delegate.skip(n);
+      byteCount += result;
+      return result;
+    } catch (IOException exception) {
+      spanStatus = SpanStatus.INTERNAL_ERROR;
+      throw exception;
+    }
+  }
+
+  @Override
+  public void close() throws IOException {
+    try {
+      delegate.close();
+    } catch (IOException exception) {
+      spanStatus = SpanStatus.INTERNAL_ERROR;
+      throw exception;
+    }
+    finishSpan();
+  }
+
+  private void finishSpan() {
+    if (currentSpan != null) {
+      if (file != null) {
+        String description = file.getName()
+          + " "
+          + "("
+          + StringUtils.byteCountToString(byteCount)
```

Comment:
whats about byte count being a `data` too?

Reply:
you mean in addition to the description or instead?

Reply:
instead, not sure if its a good idea, food for thought

Reply:
https://github.com/getsentry/develop/issues/463 we agreed to have it in the description initially, but now I see that there's a point to have under `data`, so prob. makes sense to move it there

Reply:
could be both actually, the description has a user-friendly size, in KB, MB or...
in the data, we have the real byte count

Reply:
Sounds good, will change to that 👍